httomo/data/dataset_store.py

```diff
@@ -235,7 +235,15 @@ def _create_h5_data(
         The returned data object behaves like a numpy array, so can be used freely within
         a DataSet."""
 
-        self._h5file = h5py.File(file, "w", driver="mpio", comm=comm)
+        self._h5file = h5py.File(
+            file,
+            "w",
+            driver="mpio",
+            comm=comm,
+            rdcc_nslots=1e5,
+            rdcc_nbytes=4 * (2048**2 * 16),
+            rdcc_w0=1,
+        )
```

Comment:
May I ask why this has been added in this PR?

I've seen #423 so I'm aware that improving the write speed for the dataset store when it's backed by an hdf5 file is wanting to be explored. However, as far as I can tell, that's a totally separate feature to incorporating the "watermarking" for images saved in a sweep run? (For example, I think that sweep runs don't use the reader/writer in `dataset_store.py`, they use the reader/writer from `param_sweep_store.py`).

So, unless they're connected in some way (please do say so if this is the case, I might have missed something! :sweat_smile: ), I'd suggest that the writing to the dataset store is left alone in this PR and only changes relevant to the watermark incorporation for sweep runs are added here.

Reply:
ah, good for noticing it. It is accidentality there, needs to be removed. 
I've been playing with those values a bit as @ptim0626 suggested. It seems that increasing the number of chunks larger than 1 (16 seems to be a sweat spot) increases the writing speed of hdf5 intermediate files at least 2 times. So it should be certainly in a different branch, and those values are not optimised anyway.